dagmc_stats.py

```diff
@@ -300,3 +300,184 @@ def get_coarseness(my_core, meshset, entity_ranges, geom_dim):
+                            - side_lengths[side - 1] * side_lengths[side - 1])
+                            /(2.0 * side_lengths[side] * side_lengths[side - 2])))
+    return angles
+
+
+def gaussian_curvature(my_core, vert):
+    """
+    Gets gaussian curvature of a vert
+
+    inputs
+    ------
+    my_core : a MOAB Core instance
+    vert : vertex
+
+    outputs
+    -------
+    gc : the gaussian curvature of the vert
+    """ 
+    
+    tris = my_core.get_adjacencies(vert, 2)
+    alpha_angles = 0
+    
+    for tri in tris:
+        alpha_angles += get_angles(my_core, tri, vert = vert)[0]
+    
+    gc = np.abs(2*np.pi - alpha_angles)
+    
+    return gc
+
+
+def get_local_roughness(my_core, vert):
+    """
+    Gets local roughness of a vert
+
+    inputs
+    ------
+    my_core : a MOAB Core instance
+    vert : vertex
+
+    outputs
+    -------
+    lr : the laplacian value of the vert
+    """ 
+    
+    beta_angles = []
+    d = []
+    gc_j = []
+    sum_d_gc = 0
+    
+    gc_i = gaussian_curvature(my_core, vert)
+    
+    adj_tris = my_core.get_adjacencies(vert, 2, op_type=0)
+    
+    #pair tris
+    pair_tris = []
+    tri_vert = []
+    common_vert = []
+    for tri in adj_tris:
+        tri_vert.append(my_core.get_adjacencies(tri, 0, op_type=1))
+    for tri1 in tri_vert:
+        for tri2 in tri_vert:
+            count = 0
+            same_vert = []
+            for m in range(3):
+                for n in range(3):
+                    if tri1[m] == tri2[n]:
+                        count += 1
+                        same_vert.append(tri1[m])
+            if count == 2:
+                pair = []
+                pair.append(tri1)
+                pair.append(tri2)
+                exist = False
+                for p in pair_tris:
+                    if (p[0] == tri1 and p[1] == tri2) or (p[1] == tri1 and p[0] == tri2):
+                        exist = True
+                if not exist:
+                    common_vert.append(same_vert)
+                    pair_tris.append(pair)
+    #d
+    for p in range(len(pair_tris)):
+        get_beta_angles(my_core, pair_tris[p], common_vert[p],beta_angles)
+    for x in range(0,len(beta_angles),2):
+        d.append((1/np.tan(beta_angles[x]) + 1/np.tan(beta_angles[x+1])) / 2)
+    
+    #gc
+    adj_verts = list(my_core.get_adjacencies(adj_tris, 0, op_type=1))
+    adj_verts.remove(vert)
+    for v in adj_verts:
+        gc_j.append(gaussian_curvature(my_core, v))
+    
+    for i in range(len(adj_verts)):
+        sum_d_gc += d[i] * gc_j[i]
+    lr = np.abs(gc_i - sum_d_gc / sum(d))
+    return lr
+
+
+def get_roughness(my_core, native_ranges):
+    """
+    Gets the roughness of area
+
+    inputs
+    ------
+    my_core : a MOAB Core instance
+    native_ranges : a dictionary containing ranges for each native type in the file (VERTEX, TRIANGLE, ENTITYSET)
+
+    outputs
+    -------
+    roughness: (list) the roughness for all surfaces in the meshset.
```

Comment:
Do you return roughness for each surface or each vertex? It looks like you do each vertex.